### PR TITLE
Add workflow to update esc latest-version

### PR DESCRIPTION
**Pull request by lblackstone**
### Proposed changes
Adapted from https://github.com/pulumi/docs/blob/master/.github/workflows/pulumi-cli.yml

I cut out a bunch of steps that I don't think are relevant for ESC, but that would be good to check.

### Related issues (optional)
Part of https://github.com/pulumi/esc/issues/96

### CODE REVIEW

File: .github/workflows/esc-cli.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        uses: actions/checkout@v2
+      - name: set the esc version
+        run: |
+          echo "ESC_VERSION=${{ github.event.client_payload.ref }}" >> $GITHUB_ENV
+      - name: pull-request
+        uses: repo-sync/pull-request@v2
+        with:
+          source_branch: "pulumi/${{ github.run_id }}-${{ github.run_number }}"
+          destination_branch: "master"
+          pr_title: "Regen docs esc@${{ env.PULUMI_VERSION }}"
```
Comment (pgavlin) [inline .github/workflows/esc-cli.yml:24]: should this be `ESC_VERSION`?

```suggestion
          pr_title: "Regen docs esc@${{ env.ESC_VERSION }}"
```

Comment (lblackstone) [reply .github/workflows/esc-cli.yml:24]: Yes, good catch

File: .github/workflows/esc-cli.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        run: git status && git diff
+        working-directory: docs
+      - name: commit changes
+        run: |
+          git config --local user.email "bot@pulumi.com"
+          git config --local user.name "pulumi-bot"
+          git checkout -b pulumi/${{ github.run_id }}-${{ github.run_number }}
+          git add static/
+          git add content/
+          git add data/
```
Comment (sean1588) [inline .github/workflows/esc-cli.yml:73]: this only updates the latest version in the `static/esc/latest-version` file, right? Do the content and data directories need to be added here?

File: .github/workflows/esc-cli.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        working-directory: docs
+    strategy:
+      matrix:
+        nodeversion:
+          - "18.x"
+  notify:
+    if: failure()
+    name: Send slack notification
+    runs-on: ubuntu-latest
+    needs: [pull-request, build-pulumi-cli-docs]
```
Comment (sean1588) [inline .github/workflows/esc-cli.yml:85]:
```suggestion
    needs: [pull-request, build-pulumi-esc-docs]
```

File: .github/workflows/esc-cli.yml (hunk 2 above)
Comment (lblackstone) [reply .github/workflows/esc-cli.yml:73]: Yes, just the `latest-version` file for now. I'll drop those directories.

**Review by pgavlin — APPROVED** [review]

**Pull request merged**